### DISCUSSION TITLE
Structured option for sparse grids?

### DISCUSSION THREAD

**Discussion by flotus1** [CGNS/CGNS#302] [CGNStalk]
Let's assume the mesh I need to represent via CGNS consists exclusively of Cartesian cubes. I.e. axis-aligned cubes with centroid positions that could be represented with ijk integer coordinates in 3D.
I could use the structured grid option from CGNS. But there is a caveat: the grids are usually sparsely populated. I.e. in  a block with extents (1:imax)(1:jmax)(1:zmax) I don't have (imax x jmax x zmax) elements. And setting up arrays with (imax x jmax x zmax) entries is not feasible.
Does CGNS have any special way to handle this case? Up to now, I used unstructured grids instead, with each element represented by a Hexa8 element.

**Comment by MicK7** [comment]
Using the unstructured storage seems like the optimal solution for current CGNS. Indeed you could try a full structured grid storage with iblanking but it will lead to a waste of space.
Adding a compressed spared storage to CGNS could help with your case but I'm not sure that it is general enough to be used by others.
